### DISCUSSION TITLE
Any Laravel user mind sharing their config?

### DISCUSSION THREAD

**Discussion by HassanZahirnia** [deployphp/deployer#3411] [General]
Hey I just discovered this great tool and I love it so far😊👍
But the downside I found with it is there are really few to non-existent resources on how to use this and set it up properly for projects (in my case a Laravel project).

I have a working deploy config now but I wonder if anyone mind sharing their config they use for a real world application. So far the examples and code snippets I saw didn't seem like something to be used for an actual production deployment.

For example in an actual production site, you need the user running webserver/php to be owner of your storage files. But the default Laravel recipe config does not take this into account and builds everything as the logged in user.
After lots of trials and errors I found this snippet to take care of the permission matter:

```php
set('http_user', 'www-data');
set('http_group', 'www-data');
set('writable_mode', 'chgrp');
set('writable_use_sudo', true);
set('writable_recursive', true);
set('writable_chmod_mode', '0775');
```

But I feel like this is not ideal since it asks for sudo password twice in the same deployment process.
I wonder if this can be done better, or maybe I'm missing something?

I will be glad if Laravel community could share their experience with this tool 😊🙌🌸
Thanks in advance!

**Comment by antonmedv** [comment]
Have you seen Deployer provision recipe? It can configure the whole web-server and correct permissions as well.

**Reply by HassanZahirnia** [reply]
Hey! Thanks for your quick answer! 😄
Yeah I had a quick look at it and it seems an opinionated approach to provision a freshly new server. But that's not what I'm looking for. I'm looking into using Deployer on my existing servers and projects. So the servers are already configured.

**Reply by antonmedv** [reply]
Well, it’s up to you to figure out what permissions to set)
